### PR TITLE
Draft: get_trytes API via broker

### PR DESCRIPTION
**Pull request by bingyanglin**
* Draft for the get_trytes API via broker

### CODE REVIEW

File: src/api/gettrytes.rs [hunk 1]
```diff
@@ -91,51 +88,74 @@ impl GetTrytes {
...
+                    let last_index = tx.last_index as usize;
+                    let bundle_tritbuf = decode_bytes(&tx.bundle.into_vec(), HASH_TRIT_LEN);
+                    let tag_tritbuf = decode_bytes(&tx.tag.into_vec(), TAG_TRIT_LEN);
+                    let trunk_tritbuf = decode_bytes(&tx.trunk.into_vec(), TAG_TRIT_LEN);
+                    let branch_tritbuf = decode_bytes(&tx.branch.into_vec(), TAG_TRIT_LEN);
+                    let attachment_ts = tx.attachment_ts as u64;
+                    let attachment_lbts = tx.attachment_lbts as u64;
+                    let attachment_ubts = tx.attachment_ubts as u64;
+                    let nonce_tritbuf = decode_bytes(&tx.nonce.into_vec(), NONCE_TRIT_LEN);
+                    let builder = TransactionBuilder::new()
```
Comment (wusyong) [inline src/api/gettrytes.rs:133]: There should be a `from_trit` method so you don't need to do this manually.

File: src/api/gettrytes.rs [hunk 2]
```diff
@@ -91,51 +88,74 @@ impl GetTrytes {
...
+            worker: smallbox!(worker),
+            payload: query_frame,
+        };
+        Ring::send(&self.data_center, self.replica_index, self.token, event);
+
+        while let Some(event) = rx.recv().await {
+            match event {
+                BrokerEvent::Response {
+                    giveload,
+                    query,
```
Comment (wusyong) [inline src/api/gettrytes.rs:106]: There are lots of unused variables. Are you sure they don't need to prefix with _ ?

File: src/api/gettrytes.rs [hunk 3]
```diff
@@ -91,51 +88,74 @@ impl GetTrytes {
...
+                        .with_branch(Hash::from_inner_unchecked(branch_tritbuf))
+                        .with_tag(Tag::from_inner_unchecked(tag_tritbuf))
+                        .with_attachment_ts(Timestamp::from_inner_unchecked(attachment_ts))
+                        .with_attachment_lbts(Timestamp::from_inner_unchecked(attachment_lbts))
+                        .with_attachment_ubts(Timestamp::from_inner_unchecked(attachment_ubts))
+                        .with_nonce(Nonce::from_inner_unchecked(nonce_tritbuf));
+                    // TODO: return the decoded transaction
+                    let tx = builder.build();
+                }
+                BrokerEvent::StreamStatus {
```
Comment (wusyong) [inline src/api/gettrytes.rs:152]: Just match unused event with `_ =>`

File: src/api/gettrytes.rs (hunk 1 above)
Comment (bingyanglin) [reply src/api/gettrytes.rs:133]: Do you mean `Transaction::from_trits`?

File: src/api/gettrytes.rs (hunk 2 above)
Comment (bingyanglin) [reply src/api/gettrytes.rs]: Corrected, thanks.

File: src/api/gettrytes.rs (hunk 3 above)
Comment (bingyanglin) [reply src/api/gettrytes.rs]: Corrected, thanks.

**Pull request merged**